Source/DataSources/KmlDataSource.js

```diff
@@ -1846,6 +2342,221 @@ define([
+                entitiesToIgnore.set(child.id, child);
+                recurseIgnoreEntities(child);
+            }
+        }
+
+        var cameraViewUpdate = false;
+        var lastCameraView = this._lastCameraView;
+        var camera = this._camera;
+        if (defined(camera) &&
+            !(camera.positionWC.equalsEpsilon(lastCameraView.position, CesiumMath.EPSILON7) &&
+              camera.directionWC.equalsEpsilon(lastCameraView.direction, CesiumMath.EPSILON7) &&
+              camera.upWC.equalsEpsilon(lastCameraView.up, CesiumMath.EPSILON7))) {
+
+            // Camera has changed so update the last view
+            lastCameraView.position = Cartesian3.clone(camera.positionWC);
+            lastCameraView.direction = Cartesian3.clone(camera.directionWC);
+            lastCameraView.up = Cartesian3.clone(camera.upWC);
+            lastCameraView.bbox = computeViewRectangle(camera);
+            cameraViewUpdate = true;
+        }
+
+        var newNetworkLinks = new AssociativeArray();
+        var changed = false;
+        networkLinks.values.forEach(function(networkLink) {
+            var entity = networkLink.entity;
+            if (entitiesToIgnore.contains(entity.id)) {
+                return;
+            }
+
+            if (!networkLink.updating) {
+                var doUpdate = false;
+                if (networkLink.refreshMode === RefreshMode.INTERVAL) {
+                    if (JulianDate.secondsDifference(now, networkLink.lastUpdated) > networkLink.time) {
+                        doUpdate = true;
+                    }
+                }
+                else if (networkLink.refreshMode === RefreshMode.EXPIRE) {
+                    if (JulianDate.greaterThan(now, networkLink.time)) {
+                        doUpdate = true;
+                    }
+
+                } else if (networkLink.refreshMode === RefreshMode.STOP) {
+                    if (cameraViewUpdate) {
+                        networkLink.needsUpdate = true;
+                        networkLink.cameraUpdateTime = now;
+                    }
+
+                    if (networkLink.needsUpdate && JulianDate.secondsDifference(now, networkLink.cameraUpdateTime) >= networkLink.time) {
+                        doUpdate = true;
+                    }
+                }
+
+                if (doUpdate) {
+                    recurseIgnoreEntities(entity);
+                    networkLink.updating = true;
+                    var newEntityCollection = new EntityCollection();
+                    var href = joinUrls(networkLink.href, makeQueryString(networkLink.cookie, networkLink.queryString, true), false);
+                    href = processNetworkLinkQueryString(that._camera, that._canvas, href, networkLink.viewBoundScale, lastCameraView.bbox);
+                    load(that, newEntityCollection, href)
+                        .then(getNetworkLinkUpdateCallback(that, networkLink, newEntityCollection, newNetworkLinks));
```

Comment:
We probably want an `otherwise` here that raises the `error` event and log the error to the console; otherwise we might end up with some silent failures.